Content.Server/Explosion/EntitySystems/TriggerSystem.cs

```diff
@@ -58,16 +65,62 @@ public override void Initialize()
-            SubscribeLocalEvent<TriggerOnActivateComponent, ActivateInWorldEvent>(OnActivate);
+            SubscribeLocalEvent<TriggerOnCollideComponent,     StartCollideEvent>(OnTriggerCollide);
+            SubscribeLocalEvent<TriggerOnActivateComponent,    ActivateInWorldEvent>(OnActivate);
             SubscribeLocalEvent<TriggerImplantActionComponent, ActivateImplantEvent>(OnImplantTrigger);
             SubscribeLocalEvent<TriggerOnStepTriggerComponent, StepTriggeredEvent>(OnStepTriggered);
             SubscribeLocalEvent<TriggerOnSlipComponent, SlipEvent>(OnSlipTriggered);
 
-            SubscribeLocalEvent<DeleteOnTriggerComponent, TriggerEvent>(HandleDeleteTrigger);
-            SubscribeLocalEvent<ExplodeOnTriggerComponent, TriggerEvent>(HandleExplodeTrigger);
-            SubscribeLocalEvent<FlashOnTriggerComponent, TriggerEvent>(HandleFlashTrigger);
-            SubscribeLocalEvent<GibOnTriggerComponent, TriggerEvent>(HandleGibTrigger);
+            SubscribeLocalEvent<DeleteOnTriggerComponent,           TriggerEvent>(HandleDeleteTrigger);
+            SubscribeLocalEvent<ExplodeOnTriggerComponent,          TriggerEvent>(HandleExplodeTrigger);
+            SubscribeLocalEvent<FlashOnTriggerComponent,            TriggerEvent>(HandleFlashTrigger);
+            SubscribeLocalEvent<GibOnTriggerComponent,              TriggerEvent>(HandleGibTrigger);
+            SubscribeLocalEvent<AnchorOnTriggerComponent,           TriggerEvent>(HandleAnchorTrigger);
+            SubscribeLocalEvent<SoundOnTriggerComponent,            TriggerEvent>(HandleSoundTrigger);
+            SubscribeLocalEvent<PointLightEnableOnTriggerComponent, TriggerEvent>(HandlePointLightTrigger);
+            SubscribeLocalEvent<GravityWellOnTriggerComponent,      TriggerEvent>(HandleGravityWellTrigger);
+            SubscribeLocalEvent<SingularityDistortionOnTriggerComponent, TriggerEvent>(HandleDistortionTrigger);
+        }
+
+        private void HandleDistortionTrigger(EntityUid uid, SingularityDistortionOnTriggerComponent component, TriggerEvent args)
+        {
+            var singulo = EnsureComp<SingularityDistortionComponent>(uid);
+
+            if (singulo != null)
+            {
+                singulo.Intensity = component.Intensity;
+            }
+        }
+
+        private void HandleGravityWellTrigger(EntityUid uid, GravityWellOnTriggerComponent component, TriggerEvent args)
+        {
+            var well = EnsureComp<GravityWellComponent>(uid);
+            if (well != null)
+            {
+                well.BaseRadialAcceleration     = component.RadialAcceleration;
+                well.BaseTangentialAcceleration = component.TangentialAcceleration;
+            }
+        }
+
+        private void HandlePointLightTrigger(EntityUid uid, PointLightEnableOnTriggerComponent component, TriggerEvent args)
+        {
+            if (HasComp<PointLightComponent>(uid))
+                _pointLightSystem.SetEnabled(uid, true);
```

Comment:
TryComp?

Reply:
Why? I needn't PointLightComponent variable

Reply:
![image](https://user-images.githubusercontent.com/46600554/236397825-d3b1048d-5beb-4ca0-b1cc-142a79599227.png)
Oh, but yeah, better to have it...